### PR TITLE
Escape error data in templates (closes #357, closes #355)

### LINKED ISSUE #355: Unescaped relatedSourceCode breaks reporter when decorating an error
If `relatedSourceCode` containts `<` and `>` symbols, a text between them will be treated as a HTML node when reporter parses and decorates error text. Likely there will be no suitable template in decorator to handle the text, so an exception will be thrown. So we need to escape `relatedSourceCode`.

E.g: an assertion from `ASPTests.Regression.HtmlEditor` that is passed to `relatedSourceCode`:
```
ok(html.indexOf("<B") > -1 || html.indexOf("<STRONG") > -1 || html.indexOf("BOLD") > -1);
```
As a result, reporter will try to decorate `STRONG")` tag and will throw an exception.

### LINKED ISSUE #357: Perform escaping of errors in templates
While fixing #355, I've found that we are escaping [diff description in utils](https://github.com/DevExpress/testcafe/blob/master/src/client/core/utils/dom.js#L324), that are used by assertions and actions, and [step name in runner](https://github.com/DevExpress/testcafe/blob/master/src/client/runner/runner.js#L151). In order to fix #355, I will need to escape diffs in assertions, and `relatedSourceCode` on server side. So after that we will have escaping in 4 different places, 3 on a client, and 1 on the server.

According to @kirovboris ' [comment](https://github.com/DevExpress/testcafe/issues/127#issuecomment-152106511) and @inikulin 's [comment](https://github.com/DevExpress/testcafe/issues/127#issuecomment-152125060) we are doing escaping to avoid [breaking an error template](https://github.com/DevExpress/testcafe/blob/master/src/reporter/errors/format.js#L29) parser because we use HTML markup for errors.

So I'm curious, why wouldn't we do this just right in [error templates](https://github.com/DevExpress/testcafe/blob/master/src/reporter/errors/templates.js), because here we are forming HTML markup for an error and here we can escape all including data?
\cc @inikulin @AlexanderMoskovkin @kirovboris @helen-dikareva

### PR DESCRIPTION
**Pull request by AndreyBelym**
\cc @inikulin @AlexanderMoskovkin @kirovboris

### CODE REVIEW

File: src/reporter/errors/templates.js [hunk 1]
```diff
@@ -351,3 +307,81 @@ export default {
         ${getMsgPrefix(err, CATEGORY.pageLoadError)}${err.message}
     `)
 };
+
+const UNESCAPED_ERROR_PROPERTIES = ['actual', 'expected', 'dialog', 'action', 'type', 'userAgent',
+    'pageUrl', 'isArrays', 'isObjects', 'isStrings', 'diffType'];
+
+
+function escapeError (errorType, error) {
```
Comment (inikulin) [inline src/reporter/errors/templates.js]: I don't like this approach. The good thing about templates was it' declarative nature. Now we have some black magic going on under the hood. From quick observation I don't get which properties should be escaped? BTW, do we really need to escape a lot properties? I'm still for the inline escaping in the template code.

**Comment by inikulin** [comment]
\r-

**Comment by AndreyBelym** [comment]
FPR

File: src/legacy/test-run/index.js [hunk 2]
```diff
@@ -80,7 +80,10 @@ export default class TestRun extends Session {
         }
 
         try {
-            err.screenshotPath = await this.screenshotCapturer.captureError(err);
+            var screenshotPath = await this.screenshotCapturer.captureError(err);
+
+            if (screenshotPath)
+                err.screenshotPath = screenshotPath;
```
Comment (AndreyBelym) [inline src/legacy/test-run/index.js]: I've forgot to revert it.

File: test/server/error-formatter-test.js [hunk 3]
```diff
@@ -99,18 +99,19 @@ describe('Error formatter', function () {
...
-                isArrays:          true,
+                stepName:          'Step with <tag>',
+                expected:          '"<another-tag>"',
+                actual:            '"<some-tag>"',
+                relatedSourceCode: 'eq({"<tag>": "<some-tag>"}, {"<tag>": "<another-tag>"})',
+                key:               '<tag>',
+                isObjects:         true,
                 type:              TYPE.eqAssertion,
                 userAgent:         userAgentMock,
+                screenshotPath:    '/unix/path/with/<tag>',
```
Comment (AndreyBelym) [inline test/server/error-formatter-test.js]: I've missed that assertions can have a message, we need to test it too. So I'll add a `message` field with some text (and `<tags>`, of course :P).

**Comment by AndreyBelym** [comment]
\r-

**Comment by AndreyBelym** [comment]
FPR

**Comment by inikulin** [comment]
lgtm

**Comment by kirovboris** [comment]
lgtm

**Pull request merged**